pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -359,6 +359,11 @@ func (gc *GarbageCollector) isDangling(reference metav1.OwnerReference, item *no
 		gc.absentOwnerCache.Add(reference.UID)
 		return true, nil, nil
 	}
+	if item.identity.Namespace == "" && owner.Namespace != "" {
```

Comment:
Like @jennybuckley pointed out, line 351 would be hit first and wrongly store the uid in the absentOwnerCache, so this PR does not address the issue. Please check with @liggitt if you plan to fix the issue.